### ISSUE TITLE
Menu component: Add keyboard interaction

### ISSUE THREAD

**Issue by jschuler** [patternfly/patternfly-react#5144]
Add arrow key interaction to the menu component

**Comment by tlabaj** [comment]
cc @mcarrano

**Comment by jessiehuff** [comment]
I think another point that would be worth investigating here is if we can support the two main variants of menu: 
- A menu of actions (these are our components that would semantically fit `role="menu"` like dropdown, application launcher, etc.) which I would think we could follow a pattern like [this](https://www.w3.org/TR/wai-aria-practices-1.2/examples/menu-button/menu-button-actions-active-descendant.html ) and consider using `aria-activedescendant`
- A menu of options (this is any component that would semantically fit `role="listbox"` like select)
I think it's important here to make sure that the keyboard interaction works within a screen reader and that the semantics are communicated to a user to communicate how to interact with it.

**Comment by jessiehuff** [comment]
This one can be a bit tricky since there are so many potential variations of menu. I think I would expect up/down arrow keys to navigate between menu items, right/left to navigate through the drill down aspects. In general, I'm thinking this menu component might be most like [these expectations](https://www.w3.org/TR/wai-aria-practices-1.2/#keyboard-interaction-12). 

With the fly out use case, we currently have that menu appear on hover. If it stays on hover and arrow keys are implemented, this menu should appear on focus as well for keyboard users. If using Esc on a drill down variation, focus would go to whatever toggle triggered that menu (this could be the previous menu, the original button that opened the button, etc).

I detailed breadcrumb keyboard expectations [here](https://github.com/patternfly/patternfly-react/issues/5361#issuecomment-790985061), but to summarize:
Interactive breadcrumb options would be navigated through Tab, the badge button uses enter and should be treated as a modal interaction, arrow keys within that modal, Esc to move focus back to badge button, Tab to the menu, then arrow keys should be used between the menu items.

Action or favoriting variations should be handled similarly to the Selectable favorites with left/right arrow keys. The filtering variation could be very tricky if we expect to support a similar interaction as the Typeahead Select (being able to type text into the input, see potential suggestions in the menu and then use arrow keys to navigate them). Perhaps we'd want to take a similar approach as we did with Select and use `aria-activedescendant`, but that probably needs to be tested. 

I tried to cover all of our specific use cases, but let me know if I missed any particular interaction!
